src/helpers/figures/charts/gauge_chart_rendering.ts

```diff
@@ -0,0 +1,466 @@
+  color: Color;
+  offset: number;
+}
+
+interface UnalignedRectangle {
+  topLeft: PixelPosition;
+  topRight: PixelPosition;
+  bottomRight: PixelPosition;
+  bottomLeft: PixelPosition;
+}
+
+interface Segment {
+  start: PixelPosition;
+  end: PixelPosition;
+}
+
+export function drawGaugeChart(canvas: HTMLCanvasElement, runtime: GaugeChartRuntime) {
+  const canvasBoundingRect = canvas.getBoundingClientRect();
+  canvas.width = canvasBoundingRect.width;
+  canvas.height = canvasBoundingRect.height;
+  const ctx = canvas.getContext("2d")!;
+
+  const config = getGaugeRenderingConfig(canvasBoundingRect, runtime, ctx);
+  drawBackground(ctx, config);
+  drawGauge(ctx, config);
+  drawInflectionValues(ctx, config);
+  drawLabels(ctx, config);
+  drawTitle(ctx, config);
+}
+
+function drawGauge(ctx: CanvasRenderingContext2D, config: RenderingParams) {
+  ctx.save();
+  const gauge = config.gauge;
+
+  const arcCenterX = gauge.rect.x + gauge.rect.width / 2;
+  const arcCenterY = gauge.rect.y + gauge.rect.height;
+  const arcRadius = gauge.rect.height - gauge.arcWidth / 2;
+  if (arcRadius < 0) {
+    return;
+  }
+
+  const gaugeAngle = gauge.percentage === 1 ? 0 : degreesToRadians(180 + 180 * gauge.percentage);
+
+  // Gauge background
+  ctx.strokeStyle = GAUGE_BACKGROUND_COLOR;
+  ctx.beginPath();
+  ctx.lineWidth = gauge.arcWidth;
+  ctx.arc(arcCenterX, arcCenterY, arcRadius, gaugeAngle, degreesToRadians(0));
+  ctx.stroke();
+
+  // Gauge value
+  ctx.strokeStyle = gauge.color;
+  ctx.beginPath();
+  ctx.arc(arcCenterX, arcCenterY, arcRadius, degreesToRadians(180), gaugeAngle);
+  ctx.stroke();
+
+  ctx.restore();
+}
+
+function drawBackground(ctx: CanvasRenderingContext2D, config: RenderingParams) {
+  ctx.save();
+  ctx.fillStyle = config.backgroundColor;
+  ctx.fillRect(0, 0, config.width, config.height);
+  ctx.restore();
+}
+
+function drawLabels(ctx: CanvasRenderingContext2D, config: RenderingParams) {
+  for (const label of [config.minLabel, config.maxLabel, config.gaugeValue]) {
+    ctx.save();
+    ctx.textAlign = "center";
+    ctx.fillStyle = label.color;
+    ctx.font = `${label.fontSize}px ${DEFAULT_FONT}`;
+    ctx.fillText(label.label, label.textPosition.x, label.textPosition.y);
+    ctx.restore();
+  }
+}
+
+function drawInflectionValues(ctx: CanvasRenderingContext2D, config: RenderingParams) {
+  const { x: rectX, y: rectY, width, height } = config.gauge.rect;
+  for (const inflectionValue of config.inflectionValues) {
+    ctx.save();
+    ctx.translate(rectX + width / 2 - 0.5, rectY + height - 0.5); // -0.5 for sharper lines. see RendererPlugin.drawBorders comment
+    ctx.rotate(Math.PI / 2 - inflectionValue.rotation);
+
+    ctx.lineWidth = 2;
+    ctx.strokeStyle = GAUGE_INFLECTION_MARKER_COLOR;
+    ctx.beginPath();
+    ctx.moveTo(0, -(height - config.gauge.arcWidth));
+    ctx.lineTo(0, -height - 3);
+    ctx.stroke();
+
+    ctx.translate(0, -height - GAUGE_INFLECTION_LABEL_BOTTOM_MARGIN - inflectionValue.offset);
```

Comment:
maybe some nitpick, I wonder if it would be clearer if you used `moveTo` to use explicit coordinates rather than relative ones.